advapi32.go

```diff
@@ -380,3 +399,89 @@ func ControlService(hService HANDLE, dwControl uint32, lpServiceStatus *SERVICE_
 
 	return ret != 0
 }
+
+const (
+	TOKEN_ASSIGN_PRIMARY    = 0x0001 // Required to attach a primary token to a process.
+	TOKEN_DUPLICATE         = 0x0002 // Required to duplicate an access token.
+	TOKEN_IMPERSONATE       = 0x0004 // Required to attach an impersonation access token to a process.
+	TOKEN_QUERY             = 0x0008 // Required to query an access token.
+	TOKEN_QUERY_SOURCE      = 0x0010 // Required to query the source of an access token.
+	TOKEN_ADJUST_PRIVILEGES = 0x0020 // Required to enable or disable the privileges in an access token.
+	TOKEN_ADJUST_GROUPS     = 0x0040 // Required to adjust the attributes of the groups in an access token.
+	TOKEN_ADJUST_DEFAULT    = 0x0080 // Required to change the default owner, primary group, or DACL of an access token.
+	TOKEN_ADJUST_SESSIONID  = 0x0100 // Required to adjust the session ID of an access token. The SE_TCB_NAME privilege is required.
+	TOKEN_EXECUTE           = SAR_EXECUTE | TOKEN_IMPERSONATE
+	TOKEN_READ              = SAR_READ | TOKEN_QUERY
+	TOKEN_WRITE             = SAR_WRITE | TOKEN_ADJUST_PRIVILEGES | TOKEN_ADJUST_GROUPS | TOKEN_ADJUST_DEFAULT
+	TOKEN_ALL_ACCESS        = SAR_REQUIRED | TOKEN_ASSIGN_PRIMARY |
+		TOKEN_DUPLICATE | TOKEN_IMPERSONATE | TOKEN_QUERY | TOKEN_QUERY_SOURCE |
+		TOKEN_ADJUST_PRIVILEGES | TOKEN_ADJUST_GROUPS | TOKEN_ADJUST_DEFAULT |
+		TOKEN_ADJUST_SESSIONID
+)
+
+func OpenProcessToken(hProcess HANDLE, dwDesiredAccess uint32, lpTokenHandle *HANDLE) bool {
+	*lpTokenHandle = 0
```

Comment:
Someone could presumably pass `nil` here and end up with a panic, couldn't they? I'd suggest a simple nil check to prevent that.